src/pykeen/evaluation/evaluator.py

```diff
@@ -131,19 +131,17 @@ def finalize(self) -> MetricResults:
     def evaluate(
         self,
         model: Model,
-        mapped_triples: Optional[MappedTriples] = None,
+        mapped_triples: MappedTriples,
         batch_size: Optional[int] = None,
         slice_size: Optional[int] = None,
         device: Optional[torch.device] = None,
         use_tqdm: bool = True,
         tqdm_kwargs: Optional[Mapping[str, str]] = None,
         restrict_entities_to: Optional[torch.LongTensor] = None,
         do_time_consuming_checks: bool = True,
+        training_mapped_triples: Optional[MappedTriples] = None,
```

Comment:
This is needed for filtering, I guess? Why not name it `additional_filter_triples`?

Reply:
cf. discussion in #398

Reply:
this is not the same as the ones in #398. The additional_filter_triples are going to be a third set that needs to get passed in. We could do some extra logic elsewhere to concatenate them, but I think having a consistent interface is better

Reply:
At least from the evaluation perspective, there are two sets which need to be separated:

- The evaluation triples: we compute metrics over these. In the filtered setting, we naturally use them for filtering
- Additional triples: These triples are only used for filtering.

Usually we pass the training triples as additional triples (if we do not evaluate on training - then there is no need since they are not additional). For final evaluation in the filtered setting according to Bordes, we use training *and* validation triples for filtering when we evaluate on the test set.

At least for me, the API is more clear when it reflects for what the triples are used:

- evaluation
- additional filtering

It would also make it very clear from the usage in the HPO pipeline, that we evaluate on test, and additionally filter with training and validation.

Reply:
Addressed this in 9fdcf7e7. Unlike in #398 when I asked for tests to check the new functionality works, this is maintaining current functionality with just a new parameter name